### DISCUSSION TITLE
Step custom actions

### DISCUSSION THREAD

**Discussion by smhouston** [hakimio/ngx-ui-tour#64] [Q&A] [answered]
Is there a way to perform a custom action when you get to a certain step i.e. when you get to step #2 then expand a menu in the UI and perform a server call?

**Comment by hakimio** [comment, marked as answer]
- You can listen to [TourService events](https://hakimio.github.io/ngx-ui-tour/md-menu/API#events) and perform custom actions when certain step is shown
- You can implement [custom template](https://hakimio.github.io/ngx-ui-tour/md-menu/Misc#custom-template) and handle "Prev"/"Next"/"End" button clicks however you like.

**Reply by smhouston** [reply]
I see. It would be quite nice to have functionality built in on the step setup to add a callback for code to run when that step is hit, then we can keep it encapsulated. I've managed to code around it for the time being though

**Reply by edu22snt** [reply]
Were you able to implement it?

If you did, can you show me step by step how you did it?

I need to implement something custom but I can't.
